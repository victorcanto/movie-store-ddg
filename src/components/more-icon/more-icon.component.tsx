import * as React from 'react';
import MoreVert from '@mui/icons-material/MoreVert';
import { Badge, Tooltip } from '@mui/material';
import { selectCartQuantity } from '../../store/cart/cart.selector';
import { selectFavoritedQuantity, useAppSelector } from '../../store';

export function MoreIcon() {
	const cartQuantity = useAppSelector(selectCartQuantity);
	const favoritedQuantity = useAppSelector(selectFavoritedQuantity);

	const total = cartQuantity + favoritedQuantity;

	return (
		<Tooltip title='Mais' data-testid='more-icon'>
			<Badge badgeContent={total} color='secondary'>
				<MoreVert data-testid='more-icon' />
			</Badge>
		</Tooltip>
	);
}
