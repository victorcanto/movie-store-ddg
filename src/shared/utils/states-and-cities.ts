import statesAndCities from '../data/estados-cidades.json';

export const getBrazilianStates = () =>
	statesAndCities.estados.map(({ nome }) => nome);

export const getBrazilianCitiesByState = (stateName: string) =>
	statesAndCities.estados.find(({ nome }) => nome === stateName)?.cidades;
