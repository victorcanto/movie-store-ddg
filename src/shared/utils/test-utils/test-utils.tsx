import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { setupStore, AppStore, RootState } from '../../../store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>;
	store?: AppStore;
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		store = setupStore(preloadedState),
		...renderOptions
	}: ExtendedRenderOptions = {},
) {
	function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
		return (
			<Provider store={store}>
				<BrowserRouter>{children}</BrowserRouter>
			</Provider>
		);
	}
	return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';

export { renderWithProviders as render };
