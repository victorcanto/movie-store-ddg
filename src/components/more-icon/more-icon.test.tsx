import React from 'react';
import { render } from '../../shared/utils/test-utils';
import { MoreIcon } from './more-icon.component';

test('snapshot of the base state', () => {
	const { asFragment } = render(<MoreIcon />);
	expect(asFragment()).toMatchSnapshot();
});
