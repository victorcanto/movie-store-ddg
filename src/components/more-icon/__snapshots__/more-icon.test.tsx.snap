// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of the base state 1`] = `
<DocumentFragment>
  <span
    aria-label="Mais"
    class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
    data-mui-internal-clone-element="true"
    data-testid="more-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="more-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
    <span
      class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary BaseBadge-badge BaseBadge-invisible css-fu0map-MuiBadge-badge"
    >
      0
    </span>
  </span>
</DocumentFragment>
`;
